### ISSUE TITLE
regression: genjavadoc (and browse) failing in 2.12.x build because of changes to Java comment scanning

### ISSUE THREAD

**Issue by SethTisue** [scala/community-builds#388]
at e.g. https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/920/consoleFull we see:

```
[genjavadoc] [error] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/genjavadoc-b65848a752d4e0df7428d3fb7aee7adb1de396ec/plugin/src/main/scala-2.12/com/typesafe/genjavadoc/Comments.scala:23: value skipDocComment is not a member of Comments.this.parser.UnitScanner
[genjavadoc] [error]           super.skipDocComment()
[genjavadoc] [error]                 ^
```

I believe the cause is scala/scala@fe2d7ff6821bcd0bf413b03c1c211b488a230fbe

@adriaanm maybe you could work this one out with the genjavadoc maintainers?

**Comment by SethTisue** [comment]
oh, actually browse is failing too because of the same changeset:

```
[browse] [error]  method skipComment cannot override final member
[browse] [error]            override def skipComment(): Boolean = {
[browse] [error]                         ^
```

**Comment by SethTisue** [comment]
genjavadoc folks: @2m @szeiger @ktoso

this looks like a case where Adriaan was thinking of this stuff as internals rather than API so changing it around was fair game, but genjavadoc is using it as API

**Comment by adriaanm** [comment]
Hmm, maybe I should revert, but this is final because it needs to be tail recursive, IIRC.

**Comment by adriaanm** [comment]
https://github.com/typesafehub/genjavadoc/issues/93

**Comment by SethTisue** [comment]
fixed by #411
